python/ray/train/torch/xla/config.py

```diff
@@ -105,6 +156,11 @@ def on_training_start(
     def on_shutdown(self, worker_group: WorkerGroup, backend_config: TorchXLAConfig):
         """
         Logic ran right after training is finished.
-        This is a sanity cleanup to kill xrt server.
+        This is a sanity cleanup to kill xrt server, and to optionally
+        run neuron parallel graph compilation
         """
         worker_group.execute(_kill_xrt_server)
+
+        # Compile the extracted graphs. This must run at end of training.
+        if backend_config.neuron_parallel_compile:
```

Comment:
Any ordering preference? Kill server first and then compile or compile first??

Reply:
XRT server is required for graph extraction, but not this compilation step. So order doesn't matter.